src/js/base/runtime.js

```diff
@@ -1795,10 +1803,12 @@ function (Namespace, jsnums, codePoint, seedrandom, util) {
       if(thisRuntime.isObject(val) &&
          (thisRuntime.hasField(val, "render-reason")
           || thisRuntime.hasField(val, "render-fancy-reason"))){
-        throw new PyretFailException(val);
+        var err = new PyretFailException(val);
       } else {
-        throw new PyretFailException(thisRuntime.ffi.makeUserException(val));
+        var err = new PyretFailException(thisRuntime.ffi.makeUserException(val));
       }
+      err.stackAtThrow = new Error().stack;
```

Comment:
It turns out this isn't necessary.  Thanks.  I had included this for debugging but the `stack` field on the exception itself turned out to be sufficient.